ui/dashboardApp/index.ts

```diff
@@ -72,15 +81,38 @@ async function main() {
     if (spinner) {
       spinner.remove()
     }
-    if (!routing.isLocationMatchPrefix(auth.signInRoute)) {
+    if (!routing.isSignInPage()) {
       if (!auth.getAuthTokenAsBearer()) {
-        singleSpa.navigateToUrl('#' + auth.signInRoute)
-        return
+        singleSpa.navigateToUrl('#' + routing.signInRoute)
       }
     }
   })
 
   singleSpa.start()
 }
 
-main()
+/////////////////////////////////////
+
+function start() {
+  // the portal page is only used to receive options
+  if (routing.isPortalPage()) {
+    function handleConfigEvent(event) {
+      const { token, lang, hideNav, app } = event.data
```

Comment:
prefer to name it `redirectPath` so that it can be something complex (e.g. including the query parameter part).